test/topology_experimental_raft/test_topology_upgrade_stuck.py

```diff
@@ -0,0 +1,110 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import logging
+import pytest
+import time
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.rest_client import inject_error
+from test.pylib.util import wait_for_cql_and_get_hosts
+from test.topology.util import reconnect_driver, restart, enter_recovery_state, \
+        delete_raft_data_and_upgrade_state, log_run_time, wait_until_upgrade_finishes as wait_until_schema_upgrade_finishes, \
+        wait_until_topology_upgrade_finishes, delete_raft_topology_state, wait_for_cdc_generations_publishing, \
+        check_system_topology_and_cdc_generations_v3_consistency
+
+
+@pytest.mark.asyncio
+@log_run_time
+async def test_topology_upgrade_stuck(request, manager: ManagerClient):
```

Comment:
I think we should test a more realistic scenario.
I.e. upgrade getting stuck due to a reason that can actually happen in prod -- like nodes dying.
We could also take this opportunity to test recovery in a larger cluster, because all topology upgrade/recovery tests IIRC are tested in 3 node clusters.

Because as it is now, this test seems to provide little value compared to the tests we already have.

So how about something like this?

Start with a 5 node cluster.
Have 2 nodes die during upgrade. Maybe even 3 nodes so we lose quorum.
The coordinator should at some point get stuck and print some errors. Is it doing a barrier at any point during upgrade? Global RPC? If so, this would get closed_error exceptions.
Then do recovery, removing the dead nodes (with larger cluster, this will actually do nontrivial streaming, unlike the current recovery test which IIUC leaves a single node).

Reply:
> Start with a 5 node cluster.
> Have 2 nodes die during upgrade. Maybe even 3 nodes so we lose quorum.

For quorum loss + forcing streaming during recovery removenode, even having 4 node cluster with 2 nodes dying would be enough.

Reply:
> Is it doing a barrier at any point during upgrade? Global RPC? If so, this would get closed_error exceptions.

No, AFAIK the current migration code does not do any barriers. However, I realized that the auth migration code reads does a full scan of system.auth, which probably requires the whole cluster to be alive, so maybe it is possible to simulate a "natural" scenario where the cluster gets stuck with only one node goes down - I'll try, if it doesn't work I'll modify the test to lose quorum.

Reply:
> However, I realized that the auth migration code reads does a full scan of system.auth, which probably requires the whole cluster to be alive

I'm becoming even a bigger fan of doing a global barrier in the upgrade procedure.

> I'll try, if it doesn't work I'll modify the test to lose quorum.

BTW with https://github.com/scylladb/scylladb/pull/17590 we'd get an actual exception for quorum loss too (but we should decrease timeouts with error injection if we chose this way)